zboxcore/sdk/sdk.go

```diff
@@ -886,6 +892,43 @@ func CreateAllocationForOwner(owner, ownerpublickey string,
 	return
 }
 
+func getAllocationBlobbers(owner, ownerpublickey string,
+	datashards, parityshards int, size, expiry int64,
+	readPrice, writePrice PriceRange, mcct time.Duration) ([]string, error) {
+
+	var allocationRequest = map[string]interface{}{
+		"data_shards":                   datashards,
+		"parity_shards":                 parityshards,
+		"size":                          size,
+		"owner_id":                      owner,
+		"owner_public_key":              ownerpublickey,
+		"expiration_date":               expiry,
+		"read_price_range":              readPrice,
+		"write_price_range":             writePrice,
+		"max_challenge_completion_time": mcct,
+		"diversify_blobbers":            false,
+	}
+
+	allocationData, _ := json.Marshal(allocationRequest)
+
+	params := make(map[string]string)
+	params["allocation_data"] = string(allocationData)
+	params["max_challenge_time"] = mcct.String()
+
+	allocBlobber, err := zboxutil.MakeSCRestAPICall(STORAGE_SCADDRESS, "/get_alloc_blobbers", params, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to make SC-REST Call")
+	}
+
+	var allocBlobberIDs []string
+
+	err = json.Unmarshal(allocBlobber, &allocBlobberIDs)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal blobber IDs")
+	}
```

Comment:
There should b a call to `new_allocation_request` somewhere around here. We should add to `allocationRequest` parameters, a `blobbers` field which will be a concatenation of the preferred blobbers sent by the caller with the blobber ids returned by `get_alloc_blobbers`.